misc/python/materialize/parallel_workload/executor.py

```diff
@@ -0,0 +1,80 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+import random
+import threading
+from typing import Optional, TextIO
+
+import pg8000
+
+log: Optional[TextIO]
+lock: threading.Lock
+
+
+def initialize_logging() -> None:
+    global log, lock
+    log = open("parallel-workload-queries.log", "w")
+    lock = threading.Lock()
+
+
+class QueryError(Exception):
+    msg: str
+    query: str
+
+    def __init__(self, msg: str, query: str):
+        self.msg = msg
+        self.query = query
+
+
+class Executor:
+    rng: random.Random
+    cur: pg8000.Cursor
+    pg_pid: int
+    insert_table: Optional[int]
```

Comment:
This seems unused?

Reply:
It is only used by the insert action, let me add a comment.